components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+        }
+        return nodes;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisEdge> generateEdges() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisEdge> edges = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            String child = tempHolder.getDeviceId();
+            String parent = tempHolder.getParent();
+            if (parent.equals("") || parent.equals(" ")) {
+                continue;
+            } else {
+                edges.add(new DeviceOrganizationVisEdge(parent, child));
+            }
+        }
+        return edges;
+    }
+
+    @Override
+    public List<DeviceOrganizationNode> generateHierarchy() {
+        List<DeviceOrganizationNode> hierarchy = new ArrayList<>();
+        List<DeviceOrganizationNode> parentList = new ArrayList<>();
+        List<DeviceOrganizationNode> childList = new ArrayList<>();
+        hierarchy.add(new DeviceOrganizationNode("server"));
+        parentList.add(new DeviceOrganizationNode("server"));
+        try {
+            childList = transformMetadataHolderArray(deviceOrganizationDAOimpl.getChildrenByParentId("server"));
+        } catch (DeviceOrganizationDAOException e) {
+            String msg = "Error while retrieving immediate children of server";
+            log.error(msg, e);
+        }
+        int count = 0;
+        do {
+            parentList.clear();
+            parentList.addAll(childList);
+            if (count == 0) {
+                hierarchy.get(0).setChildren(childList);
+            } else {
+
+            }
+            childList.clear();
+            try {
+                childList = transformMetadataHolderArray(deviceOrganizationDAOimpl.getChildrenByParentId(createChildIdList(parentList)));
+            } catch (DeviceOrganizationDAOException e) {
+                String msg = "Error while retrieving children of devices: " + parentList;
+                log.error(msg, e);
+            }
+            for (DeviceOrganizationNode child : childList) {
+                String currentParentId = child.getParent();
+                for (DeviceOrganizationNode parent : parentList) {
+                    if (parent.getParent().equals(currentParentId)) {
+
+                    } else {
+                        continue;
+                    }
+                }
+            }
+            count++;
+        } while (!parentList.isEmpty());
+        return hierarchy;
+    }
+
+    // this method transforms an array of type "DeviceOrganizationMetadataHolder" to an array of "DeviceOrganizationNode"
```

Comment:
@n-jay Convert these comments to proper Javadoc comments.

Reply:
Fixed 7574f0b7ce34a9d346ce30520f8c8b422e4b5468